src/mqtt/protocol/mqtt/mqtt_quic_client.c

```diff
@@ -949,9 +949,12 @@ mqtt_quic_recv_cb(void *arg)
 	case NNG_MQTT_CONNACK:
 		nng_msg_set_cmd_type(msg, CMD_CONNACK);
 		// turn to publish msg and free in WAIT state
+		p->cparam  = nni_get_conn_param_from_msg(msg);
 		conn_param_clone(p->cparam);
-		// Clone CONNACK for connect_cb & aio_cb
-		nni_msg_clone(msg);
+		// Clone CONNACK for connect_cb & user aio cb
+		if (s->cb.connect_cb) {
+			nni_msg_clone(msg);
+		}
```

Comment:
This connect cb has been deprecated. It would be replaced with the cb in mqtt_client later.

Reply:
fine